src/app.rs

```diff
@@ -1,9 +1,15 @@
 use yew::prelude::*;
 use yewprint::{Button, IconName};
 
+#[function_component(Root)]
+pub fn root() -> Html {
+    html! {
+        <App></App>
+    }
+}
+
 pub struct App {
     dark_theme: bool,
-    link: ComponentLink<Self>,
 }
```

Comment:
1. You could/should probably rewrite App to the functional component
2. I'm not sure why you introduced this Root component??